apps/meteor/app/e2e/client/rocketchat.e2e.room.js

```diff
@@ -301,32 +376,70 @@ export class E2ERoom extends Emitter {
 		return true;
 	}
 
+	async createNewGroupKey() {
+		this.groupSessionKey = await generateAESKey();
+
+		const sessionKeyExported = await exportJWKKey(this.groupSessionKey);
+		this.sessionKeyExportedString = JSON.stringify(sessionKeyExported);
+		this.keyID = (await createSha256HashFromText(this.sessionKeyExportedString)).slice(0, 12);
+	}
+
 	async createGroupKey() {
 		this.log('Creating room key');
-		// Create group key
 		try {
-			this.groupSessionKey = await generateAESKey();
+			await this.createNewGroupKey();
+
+			await sdk.call('e2e.setRoomKeyID', this.roomId, this.keyID);
+			await sdk.rest.post('/v1/e2e.updateGroupKey', {
+				rid: this.roomId,
+				uid: this.userId,
+				key: await this.encryptGroupKeyForParticipant(e2e.publicKey),
+			});
```

Comment:
Endpoint deprecated in 6.10: https://github.com/RocketChat/Rocket.Chat/pull/32197